Cleanup docs exceptions & update deprecated resource pool mutations in documentation

petercrocker opened the pull request:
Replace deprecated IPAddressPoolGetResource and IPPrefixPoolGetResource
with InfrahubIPAddressPoolGetResource and InfrahubIPPrefixPoolGetResource
respectively, as these were deprecated in Infrahub 1.5.0.

Also cleanup all the accumulated exceptions added to vale over the last 2 years.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added INFRAHUB_GIT_USE_EXPLICIT_MERGE_COMMIT configuration to control explicit merge commits.

* **Bug Fixes**
  * Accessibility and copy fixes across docs (added image alt text, minor wording corrections).

* **Documentation**
  * Renamed public GraphQL mutations to Infrahub-prefixed names for IP address/prefix allocation.
  * Standardized link, linting and style formatting across documentation.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

petercrocker committed:
Update deprecated resource pool mutations in documentation

Replace deprecated IPAddressPoolGetResource and IPPrefixPoolGetResource
with InfrahubIPAddressPoolGetResource and InfrahubIPPrefixPoolGetResource
respectively, as these were deprecated in Infrahub 1.5.0.

petercrocker committed:
Update docs CODEOWNERS from customer-success to sa

petercrocker committed:
Fix markdownlint violations across documentation

- Add missing trailing newlines (MD047)
- Add first-line headings to files that need them (MD041)
- Add alt text to images (MD045)
- Wrap bare URLs in angle brackets or convert to markdown links (MD034)
- Remove $ prefix from CLI command examples (MD014)
- Replace non-descriptive "[here]" link text with descriptive text (MD059)
- Add markdownlint disable comments to 3 snippet files that are
  intentionally partial (imported as React components)

Files modified: 42 across docs, dev commands, test fixtures, and root

petercrocker committed:
Update linting configuration and add spelling exceptions

- Update .markdownlint.yaml to enable stricter rules now that violations
  are fixed (MD001, MD014, MD034, MD041, MD045, MD047, MD059)
- Add .markdownlintignore to exclude generated/vendored directories
- Update vale sentence-case style with additional exceptions
- Add spelling exceptions for technical terms
- Fix trailing newline in CODE_OF_CONDUCT.md

petercrocker committed:
Add per-file MD001 exceptions for release notes

Add markdownlint-disable-file MD001 comments to 24 release notes files
that use h3 headings (### Fixed, ### Added, etc.) directly after tables,
which violates the heading increment rule. Using per-file exceptions
rather than a global rule to maintain stricter linting elsewhere.

petercrocker committed:
Fix remaining bare URL markdownlint violations in release notes

Convert bare GitHub PR URLs to markdown link format and fix
documentation URL in release-1_5_0.mdx. This resolves the final
MD034 (no-bare-urls) violations that were flagged by markdownlint.

petercrocker committed:
Centralize markdownlint-cli2 configuration and ignore patterns

Create .markdownlint-cli2.yaml to centralize configuration and ignore
patterns for markdownlint-cli2. This file is automatically loaded by
both the CLI and the GitHub Action, ensuring consistent behavior.

Key changes:
- Add .markdownlint-cli2.yaml with ignores for docs/docs/reference/,
  CLAUDE.md, .venv/, node_modules/, and other non-lintable paths
- Simplify CI workflow to rely on the config file for ignores
- Update invoke tasks to use simpler glob patterns

This fixes the issue where .markdownlintignore was not being respected
by the DavidAnson/markdownlint-cli2-action in CI.

petercrocker committed:
Remove redundant .markdownlintignore file

The ignore patterns are now in .markdownlint-cli2.yaml, which is
automatically loaded by markdownlint-cli2.

petercrocker committed:
Fix MDX compilation errors caused by angle-bracket URLs

In MDX files, angle brackets are parsed as JSX elements, so URLs like
<https://example.com> cause compilation errors because the `/` character
is unexpected in JSX tag names.

Changes:
- Convert angle-bracket URLs to proper markdown links in 38 MDX files
- Use backticks for inline URL display where appropriate
- Fix backend.mdx, faq.mdx, and all release notes with demo-edge URLs

petercrocker committed:
Expand Vale and markdownlint configuration and fix lint violations

- Add comprehensive spelling exceptions for technical terms, product names,
  and snake_case identifiers used in documentation
- Streamline sentence-case exceptions to use word-level patterns instead of
  full heading matches for better maintainability
- Configure Vale to ignore docs/docs/reference/** and **/AGENTS.md files
- Add AGENTS.md to markdownlint ignores
- Fix GitHub capitalization across codebase
- Replace ambiguous "e.g." with "i.e." where appropriate
- Fix "idempotence" to "idempotency" for consistency
- Minor copy improvements in release notes

petercrocker committed:
Merge branch 'stable' into pmc-20251227-docs-cleanup

petercrocker committed:
cleanup

petercrocker committed:
Fix MDX compilation error in faq.mdx

Convert angle-bracket email address to markdown link format to avoid
MDX JSX parsing error with @ character.

petercrocker committed:
Add baseline-browser-mapping dependency to docs

dgarros commented on tasks/docs.py:
```diff
@@ -130,23 +130,29 @@ def vale(context: Context) -> None:
     if not has_markdownlint:
         print("Warning, markdownlint-cli2 is not installed")
         return
-    exec_cmd = "markdownlint-cli2 '**/*.{md,mdx}' '#**/node_modules/**' '#.venv/**'"
```
We need to revert this change

dgarros commented on tasks/docs.py:
```diff
@@ -130,23 +130,29 @@ def vale(context: Context) -> None:
+    Uses .markdownlint-cli2.yaml for configuration and ignore patterns.
+    """
     print(" - [docs] Format code with markdownlint-cli2")
-    exec_cmd = "markdownlint-cli2 '**/*.{md,mdx}' '#**/node_modules/**' '#.venv/**' --fix"
```
same here

dgarros commented on .markdownlint-cli2.yaml:
```diff
@@ -0,0 +1,26 @@
+
+# Files to ignore (replaces .markdownlintignore)
+ignores:
+  - ".claude/**"
```
This list looks wrong to me, unless I'm missing something we want the linter to run on the AGENTS and CLAUDE files

dgarros commented on .markdownlint-cli2.yaml:
```diff
@@ -0,0 +1,26 @@
+# https://github.com/DavidAnson/markdownlint-cli2
+
+# Use the existing markdownlint configuration
+config:
```
Could we merge these 2 files instead of extending `.markdownlint.yaml`

petercrocker committed:
Merge branch 'stable' into pmc-20251227-docs-cleanup

petercrocker committed:
Consolidate markdownlint config and scope to docs/docs/

- Merge .markdownlint.yaml settings into .markdownlint-cli2.yaml
- Add globs to only lint .md and .mdx files in docs/docs/
- Simplify ignores to only exclude docs/docs/reference/
- Delete redundant .markdownlint.yaml

petercrocker committed:
Remove explicit globs from markdownlint invocations

The globs are now defined in .markdownlint-cli2.yaml, so remove
the overriding globs from:
- CI workflow (.github/workflows/ci.yml)
- Invoke tasks (tasks/docs.py)

This ensures markdownlint only scans docs/docs/**/*.{md,mdx} as
configured in the central config file.

petercrocker committed:
Add markdownlint-cli2 to docs devDependencies

- Add markdownlint-cli2 (>=0.17.0) to docs/package.json devDependencies
- Update tasks/docs.py to fail with helpful error message instead of
  silently skipping when markdownlint-cli2 is not installed

petercrocker committed:
Pass markdownlint globs explicitly via CLI

The globs in .markdownlint-cli2.yaml are not consistently respected
by markdownlint-cli2 when run via CLI or GitHub Action. Pass them
explicitly to ensure only docs/docs/**/*.{md,mdx} files are scanned.

petercrocker committed:
Regenerate configuration reference docs
